lib/modules/datasource/metadata.ts

```diff
@@ -170,3 +169,51 @@ export function addMetaData(
     }
   }
 }
+
+export function setSourceUrlToHomepage(dep: ReleaseResult): void {
+  if (!dep?.homepage) {
+    return;
+  }
+  const platform = detectPlatform(dep.homepage);
+  if (platform !== 'github' && platform !== 'gitlab') {
+    return;
+  }
+  dep.sourceUrl = dep.homepage;
+  if (shouldDeleteHomepage(dep.sourceUrl, dep.homepage)) {
+    // remove homepage if its not a link to a path in a github/gitlab repo.
+    delete dep.homepage;
+  }
+}
+
+export function shouldDeleteHomepage(
+  sourceUrl: string,
+  homepage: string
+): boolean {
+  const massagedSourceUrl = massageUrl(sourceUrl);
+  if (massagedSourceUrl === homepage) {
+    return true;
+  }
+  const sourceUrlParsed = parseUrl(massagedSourceUrl);
+  if (is.nullOrUndefined(sourceUrlParsed)) {
+    return false;
+  }
+  const homepageParsed = parseUrl(homepage);
+  if (is.nullOrUndefined(homepageParsed)) {
+    return false;
+  }
+  // if urlDepth is less than or equal to 2 then url is not
+  // a link to a path in the repo.
+  // this case handles these kinds of links:
+  //   github.com/org
+  //   github.com/org/
+  //   gitub.com/org/repo
+  //   github.com/org/repo/
+  if (urlPathDepth(homepage) <= 2) {
+    return true;
+  }
+  let hPath = homepageParsed.pathname;
+  if (hPath.charAt(hPath.length - 1) === '/') {
+    hPath = hPath.substring(0, hPath.length - 1); // remove last slash
+  }
+  return hPath === sourceUrlParsed.pathname;
```

Comment:
i think still think we can remove `homepage` when both urls (with trim trailing slash) are the same. The changelog code will do a platform check again, so this whole check doesn't matter at all and should be simplified.

Renovate will also fallback to `sourceUrl` for the PR homepage link if `homepage` is missing.

Reply:
Removed the check and will take note of this for the future.